### PR TITLE
Implemented stdproc::lists

### PR DESCRIPTION
**Pull request by Kayla-Kwock**
Implemented functions and test cases for lists. Revised the `reverse` function in `system::core::lists`. Changed    `pairs_and_lists.cpp` to utilize `namespace shaka` to reduce clutter and improve readability.

**Please check for any issues/mistakes, particularly in memory management**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -83,7 +83,7 @@ set(SOURCE_FILES
         src/shaka_scheme/system/base/Vector.cpp
         )
 
-add_library(${SHAKA_SCHEME_LIBRARY_NAME} SHARED ${SOURCE_FILES})
+add_library(${SHAKA_SCHEME_LIBRARY_NAME} SHARED ${SOURCE_FILES} src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp)
```
Comment (CinchBlue) [inline CMakeLists.txt]: Eek! This is the wrong place to put the source file. It should instead be above in the long list and appended there.

File: src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+}
+
+//(list? ...)
+Args is_list(Args args) {
+  if (args.size() != 1) {
+    throw InvalidInputException(8,
+                                "list?: Invalid number of arguments for "
+                                    "procedure");
+  }
+  return {create_node(Data(Boolean(
```
Comment (CinchBlue) [inline src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp]: This is wrong. The null list is also a list but it is not a pair.

File: src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+    }
+
+    // For the value of the given number
+    //  Add a null list at the end
+    for (int i = 0; i < args[0]->get<Number>(); i++) {
+      list_args.push_back(create_node(Data()));
+    }
+  }
+    // Else when 2 arguments are given
+  else {
```
Comment (CinchBlue) [inline src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp:154]: Make sure to make this invalid for cases of 3 arguments and more. I may be wrong, so reference R7RS here closely.

**Review by CinchBlue — CHANGES_REQUESTED** [review]
Overall, this is pretty good. However, you may want to add more test cases, and perhaps manually test it yourself in Shaka Scheme's REPL. You can create a `integ-core-lists.cpp` file if you so desire to take Shaka Scheme's parsing and evaluation systems and evaluate expressions

File: CMakeLists.txt (hunk 1 above)
Comment (CinchBlue) [reply CMakeLists.txt]: Make sure that the parenthesis stays on its own line -- it should be alone on line 85 after you are done placing the source file on line 84.

File: src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp (hunk 3 above)
Comment (Kayla-Kwock) [reply src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp:154]: On line 130 of pairs_and_lists.cpp, it throws an exception when there are more than 2 arguments or no arguments given.

File: CMakeLists.txt (hunk 1 above)
Comment (Kayla-Kwock) [reply CMakeLists.txt]: Good point! Changes made.

File: src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp (hunk 2 above)
Comment (Kayla-Kwock) [reply src/shaka_scheme/runtime/stdproc/pairs_and_lists.cpp]: Changed!

**Review by CinchBlue — APPROVED** [review]

**Pull request merged**